src/factory.js

```diff
@@ -46,11 +41,19 @@ class Factory {
     /** @type ControllerOptions */
     this.opts = merge(defaults, options)
 
+    // conditionally include ipfs based on which type of daemon we will spawn
+    if (this.opts.type === 'js' || this.opts.type === 'proc') {
+      this.opts.ipfsModule = {
```

Comment:
This will overwrite `ipfsModule` that the user has passed in. We should only set this if `this.opts.ipfsModule` is not yet set.

Reply:
Good shout, have pushed a fix

Reply:
this is just to avoid throwing when spawning go controllers?

Reply:
Yes, when your project is only going to use `go-IPFS` there's not much point installing `js-IPFS` as well.

Reply:
this still fails when bundling (or at least warns depending on the bundler) and if the user wants to use js or proc they are forced to specify it in the factory constructor or else they get an error when calling `.spawn({type:'proc'})`

Reply:
I've moved that clause to `.spawn` so it's run against the merged options